### ISSUE TITLE
Coin costs while exchanging crypto against each other

### ISSUE THREAD

**Issue by boytunp** [bitsofwinter/cryptotaxsweden#2]
Please correct me if I am wrong but as I understand from glancing at the code:

Let's say you bought BTC with SEK then bought LTC with that BTC then sold that LTC for SEK.
The cost basis calculation always adds the column "value in SEK" price to the average. In the above scenerio The cost for buying that LTC is taken from the cointracking database in SEK on the trade day. But the actual cost is the SEK value of BTC you bought. Similar example is given on the [https://www.skatteverket.se/privat/skatter/vardepapper/andratillgangar/kryptovalutor.4.15532c7b1442f256bae11b60.html](url)  third example down on the page.
![screenshot 2018-04-15 at 23 09 46](https://user-images.githubusercontent.com/23097617/38783439-4aad42ac-4102-11e8-9418-ab4e9ae73a3c.png)

**Comment by bitsofwinter** [comment]
Jag tror Skatteverkets beskrivning är det som scriptet gör, men så som de skriver är det lite otydligt. Det de skriver är att omkostnaddsbeloppet för de BTC som köpts är 10000kr men sedan är försäljningspriset för dessa BTC värdet i SEK av de LTC som köptes. Jag tolkar detta som att omkostnadsbeloppet för LTC är samma värde i SEK för dessa LTC vid växlingstillfället, men detta står ju inte yttrycklingen eftersom de inte talar om vad som skall göras när dessa LTC säljs.

**Comment by bitsofwinter** [comment]
Do you believe this is still an issue? Or should I close the issue?

**Comment by mburazin** [comment]
I was thinking on this, and I think also that this is not handled properly currently. 
I see it like this:

- when buying a coin, the price that you bought it for (buying price) is what you gave for it, and what you gave corresponds to the selling price of the coin you sell at that moment.
- When selling a coin, the price that you sell it for, is what you received for it, and what you receive is the price of the coin bought at that moment...

I don't know if this sounds understandable, but to me it makes sense if you think about it. I even made this modification in the code in my private repo and will use that to calculate my tax.

**Comment by bitsofwinter** [comment]
Yes, I think what you are describing is how the script currently does the computation. I changed it to use a method similar to this after discussing with someone about it, might have been with you.

**Comment by mburazin** [comment]
Hm... Are you sure it's updated like you said ?

I still see this code: 
```
        if trade.type == 'Trade':
            buy_coin = get_buy_coin(trade)
            sell_coin = get_sell_coin(trade)

            if trade.sell_coin == native_currency:
                value_sek = trade.sell_value
            else:
                value_sek = trade.buy_value

            if buy_coin:
                buy_coin.buy(trade.buy_amount, value_sek)
            if sell_coin:
                tax_event = sell_coin.sell(trade.sell_amount, value_sek)
                if trade.date >= from_date:
                    tax_events.append(tax_event)
```

In my opinion should be like this: 
```
        if trade.type == 'Trade':
            buy_coin = get_buy_coin(trade)
            sell_coin = get_sell_coin(trade)

            if buy_coin:
                buy_coin.buy(trade.buy_amount, trade.sell_value)
            if sell_coin:
                tax_event = sell_coin.sell(trade.sell_amount, trade.buy_value)
                if trade.date >= from_date:
                    tax_events.append(tax_event)
```

Do you agree ?

**Comment by metalmetalmetal** [comment]
I talked to Skatteverket today to see whether they could clear up a potential confusion regarding their examples.
In one of their examples, they're talking about Samira and her crypto trades in a general form.
There's, however another, more specific example about trading bitcoin and ether. It's the example where Dagny is taking the role as a trader.

I was confused about whether these two had any correlation to eachother and finally got an answer that the example with Dagny is an elaboration and more concrete example of what's described in Samiras case.
I actually got this confirmed by two representatives, so I guess it means it's pretty solid validation.
I don't know if that helps you any, but if there was any doubt as to if those where the same kind of situations, just described differently, then yes. That's exactly what those are.

**Comment by bitsofwinter** [comment]
mburazin: The issue if you write it like that is then you can get a spread when doing an exchange between two cryptos, i.e. what you sold and what you bought will not be worth the same. But then again, I'm not sure what is correct as I don't think Skatteverket has specified this clearly.

metalmetalmetal: Hmm... as far as I can remember the example with Dagny doesn't say what the cost basis for the ether should be, without that information it isn't clear how a crypto-crypto trade should be handled.

**Comment by bitsofwinter** [comment]
But if you did get information about the cost basis after a crypto-crypto trade then it would be great info. Please share if you did get that info! :-)

**Comment by mburazin** [comment]
So this is the quote that tells us what the selling price is for the sold bitcoin when doing BTC -> ETH exchange:
> Försäljningspriset för de bortbytta bitcoinen blir värdet i svenska kronor av de ether hon har fått vid växlingstillfället

So, this means that the selling price of BTC is the value (in SEK) of the ETH bought at time of trade, since that is the value you received for the BTC you have sold. 
Wouldn't it be logical to conclude that the buying price of ETH in this case is what the person actually payed for it, which is in fact the market value of BTC he/she disposed of (in SEK) ?

@metalmetalmetal, what we need from Skatteverket (since they say what is the selling price for disposed coins) is what is the buying price of other coin in the exchange (in this case ETH) since it's only when buying a coin that changes the average cost-basis of the bought coin.

The question is, do they even know?

**Comment by bitsofwinter** [comment]
Not sure what is the logical conclusion here. Let's just agree that the information isn't available yet.

Another thought though, if I was to use the same argument when exchanging SEK for BTC it sounds a bit more strange, i.e. that the buying price of SEK would be the marked value of the BTC sold and the selling price of SEK should be the value of the BTC received, which would mean that capital gains would apply also to the SEK involved in the trades.

This issue is confusing, let's hope Skatteverket can be super duper extra clear here soon. I would really prefer that they actually updated the home page with a clear example though. What they say when calling them on the phone could be wrong or change.

**Comment by boytunp** [comment]
In the US they allow you to use chose from a variety of methods FIFO, FİLO,
, average etc. to calculate cost basis. So it would not be the end of the
world  if one would use any differently interpreted methods as in this
discussion to decide what is the the buying price of a coin when
calculating the cost basis.

In plain English it's a grey area and open to interpretation. This means
you can close the issue :)

**Comment by mburazin** [comment]
For now, maybe a good strategy for the paranoid kind is to choose whatever generates more tax for you at the end, especially if the difference isn't so great between the two. That's why I think I will go with the way the script calculates it now after all, since that's what generates a bit more tax for me.

And I agree, we can speculate all we want, but the fact is that there is no official information about this yet.
Thanks for all the comments.

**Comment by metalmetalmetal** [comment]
I just got off the phone with Skatteverket about how to do calculations. I've gotten a bit better understanding of the procedure (which is _insanely_ cumbersome), but I'm also a bit confused of a few things, which correspond to the value increase of the crypto you are trading with, at the point of transaction. It's not so much that it _matters_ but _how_ it matters when doing the calculations.

# FIFO vs LIFO vs average
Now, when it comes to the FIFO, LIFO and whatnot it's always "cost average"(?)/omkostnadsbelopp as the base for the transaction.
The guy explicitly said there is no FIFO or LIFO or anything like that.

As an example was
* You own 4.8 bitcoins (that you've acquired over a period of time) that you bought for the combined worth of 20000
* You buy 1 Monero for the amount of 0.048 bitcoins
* Your omkostnadsbelopp is now 1/100 of your owned bitcoins. This gives you 20000/100 = 200 SEK as omkostnadsbelopp, for this transaction.

And this you repeat for every transaction, but e.g. in the other direction when you go from Monero back to Bitcoin.

So, does this example validate the script and how it does calculations? And I'm meaning in general, not only LIFO/FIFO/omkostnadsbelopp.

# Sourced costs
I mentioned that the reports I get, use an averaged calculation of the worth of each coin and the guy I talked to said that he sees very little reason for why Skatteverket would object to that, because they can't point to a place where you would verify SEK prices of bitcoin over time.

I'm going to call back in a while again with a couple of follow up questions. I saw your question, @mburazin 
Is there anything else that anyone would want me to bring up?

**Comment by bitsofwinter** [comment]
Transfering the cost basis from the previous coin like that can't be right. The capital gains realized for the coin being sold is based on the current value in SEK for the coin being bought, then it seems extremely strange that the cost basis for the coin being bought should be based on something else than a similar value in SEK.

If this was true I could see this being abused both for getting extremely high and extremely low taxes.

You could on purpose realize capital losses over and over, if you bought something that has depreciated in value, by just exchanging those coins between two cryptos over and over. They must be really confused at Skatteverket or you understood them wrong.

**Comment by mburazin** [comment]
Excuse me for the following long message, but I just want to make sure we are all on the same page, and maybe someone makes use of the following information...

I think it's important to understand that this term "omkostnadsbelopp" is used in three different contexts:

1) Totalt omkostnadsbellop (total cost) - in your example it's 20000 SEK which you spent in total while buying BTC. It consists of all the buying prices of BTC for each buy summed together.
2) Genomsnittligt omkostnadsbelopp (average cost) - you get it by dividing (total cost) / (amount of BTC), in your example 20000/4.8 = 4166,67 SEK
3) Använt omkostnadsbelopp (used cost for the sale you did, only when you sell) - you get it by multiplying (amount of BTC sold) * (average cost) in your example it's 0.048*4166,67 = 200 SEK

So now that you bought your Monero and sold BTC for it, your total omkostnadsbelopp for BTC has changed from 20000 SEK - 200 SEK = 19800 SEK. The genomsnittligt omkostnadsbelopp has stayed the same, since 19800/(4.8 - 0.048) = 4166,67 SEK. This means for the sales of BTC, your average cost of coin is not changed.

It's only when you buy the coin, is that the average cost (genomsnittligt omkostnadsbelopp) changes and it depends on the buying price (like it did when you bought BTC for SEK).
And that's why it's important to know what is actually the buying price of the coin is when going from coin to coin, because that is what changes the average cost (Genomsnittligt omkostnadsbelopp), which affects every subsequent sale of that coin, which in turn affects the accumulated tax amount.

Btw., The tax for each sale is calculated by subtracting (coin sale price) - (total cost). In your example of selling the BTC for monero your sell price for BTC would be the market value of Monero (as per Skatteverkets information).
But in order to now calculate all the different "costs" (omkostnadsbelopp) for Monero, you need to know what the buying price of that Monero was in SEK. So, is it the market price of Monero at the time of trade, or is it the market price of BTC you sold at the time of trade ?
It's easy when you go from SEK to Coin, because you know the SEK amount directly, but when you go from Coin1 to Coin2, what amount did you buy the Coin2 for ??

I hope this wasn't too long and helps understand the dilemma better :-)

**Comment by mburazin** [comment]
I also agree with @bitsofwinter more and more on this :-) 
But still an official statement from Skatteverket would be nice thing to have

**Comment by bitsofwinter** [comment]
@mburazin Great clarification and I am on the same page as you on this one! Let's see if Skatteverket can clarify this. Until they do I guess I will use the current computation method, as you say it generates slightly more tax and that should be a safer option to stay out of trouble when the information is unclear.

**Comment by metalmetalmetal** [comment]
@bitsofwinter 
I'm fairly sure I understood them wrong, or at least, I couldn't dispute their reasoning because I'm historically not financially involved other than this specific instance, in terms of doing taxes that doesn't involve sending in whatever report Skatteverket sends out and pushing the "ok"-button on that.
I am a newbie when it comes to this, and will most likely have to hire an accountant to get these things in order, and/or verified.
So, I understood the guy as saying "After your first purchase of Monero, you used the omkostnadsbelopp from your previously bought BTC. Now you have/can calculate an omkostnadsbelopp for Monero, from your first purchase, so you use that omkostnadsbelopp when selling back to BTC."
Is this where you're in disagreement?

@mburazin 
> Btw., The tax for each sale is calculated by subtracting (coin sale price) - (total cost). In your example of selling the BTC for monero your sell price for BTC would be the market value of Monero (as per Skatteverkets information).
> But in order to now calculate all the different "costs" (omkostnadsbelopp) for Monero, you need to know what the buying price of that Monero was in SEK. So, is it the market price of Monero at the time of trade, or is it the market price of BTC you sold at the time of trade ?
> It's easy when you go from SEK to Coin, because you know the SEK amount directly, but when you go from Coin1 to Coin2, what amount did you buy the Coin2 for ??

I'ma call them and see what they say about this.

**Comment by mburazin** [comment]
@metalmetalmetal, so qouting what you just wrote to @bitsofwinter:

> So, I understood the guy as saying "After your first purchase of Monero, you used the omkostnadsbelopp from your previously bought BTC. Now you have/can calculate an omkostnadsbelopp for Monero, from your first purchase, so you use that omkostnadsbelopp when selling back to BTC.

Does this mean we have even a third option as a possibility of a buying price for the bought coin (Monero).

To summarize, now I even see three possibilities in the following example:
You exchange BTC -> Monero:

1) Buying price of Monero is the market value of Monero at the time of the trade
2) Buying price of Monero is the market value of BTC at the time of the trade
3) Buying price of Monero is the "used cost" (Använt omkostnadsbelopp) of the BTC amount you exchanged (sold) to get the Monero in that trade

Possibility number 3) actually makes a lot of sense to me when I think about it. Or is my brain just completely fried from all of this? 

Maybe we could try to modify the script and try what would the calculation of tax due be in the third case as well and see how it differs from the first two.

Any opinions on this?

**Comment by mburazin** [comment]
Just to clarify, options 1 and 2 we had discussed before, but it seems there is option 3 which we haven't thought about

**Comment by bitsofwinter** [comment]
Option 3 was what I meant could be abused if the selling price for the BTC is the market value of the Monero. If option 3 is even to be considered then I think the selling price for the BTC must be the used cost for that BTC as well, but that would in the end mean that only fiat-crypto trades would be taxed in the end and I don't think that's what Skatteverket had in mind. But then again, my mind might be fried as well.

**Comment by bitsofwinter** [comment]
And with abuse I mean for example:

1. Buying 1 BTC for 10000 SEK.
2. Bitcoin drops to 5000 in value.
3. Exchange 1 BTC to X ETH => capital loss of 5000. Cost basis for X ETH is still 10000 SEK.
4. Exchange X ETH to 1 BTC => capital loss of 5000. Cost basis for 1 BTC is still 10000 SEK.
5. repeat infinitely until you have 0kr tax for this year to pay. ;-)

**Comment by mburazin** [comment]
@bitsofwinter You are absolutely right! Thanks for making it clear so we can eliminate the option 3 from consideration :smiley: 
Now we are back where we were - option 1 or 2.

**Comment by OriginalMadman** [comment]
@bitsofwinter not really, once you incur the capital loss that's the new value, the cost basis for X ETH will be 5000 SEK - never 10000 SEK (in this example).

**Comment by bitsofwinter** [comment]
@OriginalMadman I know, the point of the example was to illustrate that option 3 wasn't really an option.

**Comment by Greveyaro** [comment]
Looks like Cointracking supports average cost basis now. 
They also have adjusted cost basis, and they mention Sweden, huh? 

Scroll down to 22-12-2018

https://cointracking.info/new_features.php

**Comment by ABEC7** [comment]
These are the results i get using the different calcualtion methods at Cointracking;

AVCO at Cointracking results in x SEK in taxes (according to Cointracking's tax report, ie. before adjusting deduction of only 70 % of losses)

ACB gives roughly 2x SEK in taxes before adjusting deduction of only 70 % of losses

ACB gives roughly 3x SEK in taxes after adjusting deduction of only 70 % of losses

Cryptotaxsweden's script gives rougly 6x SEK in taxes.

**Comment by bitsofwinter** [comment]
ABEC7, that is quite a large difference. I don't know any details about how cointracking do their average cost basis calculation, just that it doesn't take the 70% deduction of losses into account and because of that it gives lower tax. I don't know if they have tried to make their calculations compatible with swedish laws or not.

Have you tried to analyze whether their calculations are correct? I've heard from another user that he thought they were not correct at all for swedish purposes.

**Comment by bitsofwinter** [comment]
Just did a test now and if I manually adjust the loss to 70% then I get something which is very close to what the script outputs if I use AVCO on cointracking.

**Comment by bitsofwinter** [comment]
But doing the same for 2017 gives a larger difference. I would need to dig further into the details to know why. For now I'm trusting the person who told me that the calculations on cointracking are not correct, he seemed to have spent a lot of time on analyzing the situation and has also written his own tool to calculate tax and came to very similar results as my script did.

**Comment by ABEC7** [comment]
No, i haven't analyzed their calculations. Based on the fact that they don't highlight AVCO or any of the other methods as "Sweden compatible" i doubt they have done a proper check. I'm going with the tax generated by your script, even if this results in me paying some extra taxes I'd rather do that than risk paying penalties and such.

**Comment by xrper21** [comment]
I need help in submitting a K4 even though it’s past the deadline they still want me to submit it. I’m a newbie and I don’t know what to do . Is cointracking good enough to use for k4

**Comment by bitsofwinter** [comment]
Yes, cointracking in combination with using this script will generate a K4 for you, the instructions are in the README of the script. There are also other services available now that can generate a K4 but I haven't tried them so don't know how well they work. Please move this discussion to a separate issue though.
